src/engine/surface_collision.c

```diff
@@ -462,35 +468,38 @@ struct Surface *find_water_floor_from_list(struct SurfaceNode *surfaceNode, s32
     f32 curHeight = FLOOR_LOWER_LIMIT;
     f32 bottomHeight = FLOOR_LOWER_LIMIT;
     f32 curBottomHeight = FLOOR_LOWER_LIMIT;
+    f32 buffer = FIND_FLOOR_BUFFER;
 
     // Iterate through the list of water floors until there are no more water floors.
+    // SURFACE_NEW_WATER_BOTTOM
     while (bottomSurfaceNode != NULL) {
         surf = bottomSurfaceNode->surface;
         bottomSurfaceNode = bottomSurfaceNode->next;
 
         // skip wall angled water
-        if (absf(surf->normal.y) < NORMAL_FLOOR_THRESHOLD) continue;
+        if (surf->type != SURFACE_NEW_WATER_BOTTOM || absf(surf->normal.y) < NORMAL_FLOOR_THRESHOLD) continue;
 
-        if (surf->type != SURFACE_NEW_WATER_BOTTOM || !check_within_floor_triangle_bounds(x, z, surf)) continue;
+        if (!check_within_bounds_y_norm(x, z, surf)) continue;
 
         curBottomHeight = get_surface_height_at_location(x, z, surf);
 
-        if (curBottomHeight < y - FIND_FLOOR_BUFFER) {
```

Comment:
Was the change from negative to positive here intentional?

Reply:
yup! it was a bit awkward before, you'd fall out too high